src/cpp/session/modules/SessionRmdNotebook.R

```diff
@@ -759,83 +768,89 @@ assign(".rs.notebookVersion", envir = .rs.toolsEnv(), "1.0")
          warning("Failed to parse chunk options in body:\n", e)
       })
    }
-   
+
    # ensure that fields we expect to be logical are actually logical
    # (sanitize invalid inputs here since they can break notebook execution)
    fields <- list(
       warning = TRUE,
       message = TRUE,
       error   = FALSE
    )
-   
+
    .rs.enumerate(fields, function(key, default) {
-      
+
       if (is.null(opts[[key]]))
          return()
-      
+
       opts[[key]] <<- tryCatch(
          as.logical(opts[[key]]),
          error = function(e) default
       )
-      
+
    })
-   
+
    # if this is a SQL chunk, try to provide a default connection
    # TODO: arguably, we should be using `knitr::opts_chunk$get()`
    # to provide defaults for chunk options, but it's not clear if
    # that's always the correct choice, so we just make the smaller
    # piece-meal decision here
    if (opts$engine %in% c("mysql", "psql", "sql")) {
-      
+
       # get current connection
       conn <- .rs.nullCoalesce(
          opts$connection,
          knitr::opts_chunk$get("connection")
       )
-      
-      # our chunk execution tools expect a character connection,
-      # so use a temporary variable to store the connection value
+
+      # our chunk execution tools expect connection to be a character naming a variable in the environment
+      # if conn is provided as a symbol, pass connection the actual variable name
+      # if conn is provided as a DB connection, use a temporary variable
+      # to store the connection value and pass the temp variable name
       if (!is.null(conn) && !is.character(conn)) {
-         .rs.setVar("chunkConnection", conn)
-         opts$connection <- ".rs.chunkConnection"
+         if (is.name(conn)) {
+            opts$connection <- as.character(conn)
+         } else {
+            .rs.setVar("chunkConnection", conn)
+            opts$connection <- ".rs.chunkConnection"
+         }
```

Comment:
basically, if we're using in-body chunk options in csv-format (tag=value), then we call `knitr:::parse_params` to parse those chunk options for us. which will end up making `conn` a symbol (e.g. `conn=as.name("db")`), so calling as.character here allows us to set `opts$connection <- "db"